modules/tinymce/src/core/main/ts/api/ThemeManager.ts

```diff
@@ -18,7 +18,7 @@ export type Theme = {
   execCommand? (command: string, ui?: boolean, value?: any): boolean;
   destroy? (): void;
   init? (editor: Editor, url: string, $: DomQueryConstructor);
-  renderUI? (): { iframeContainer?: HTMLIFrameElement; editorContainer: HTMLElement; api?: EditorUiApi };
+  renderUI? (): { iframeContainer?: HTMLIFrameElement; editorContainer: HTMLElement; api?: Partial<EditorUiApi> };
```

Comment:
Are we okay with not all methods here being provided? And where is this implemented — did it need to change there also? Also, if this is `Partial` now, can we take out the optional?

Reply:
I assume this is just something we missed last time? As if the API isn't specified, then it'll just be a noop (as noted in `Editor.ts`).

I do wonder if we should make it be an implement all, or nothing, but that'd cause issues with adding new APIs as it'd be a breaking API change for anyone who has custom themes. As such I'm happy with making this a `Partial`.

Reply:
I made it optional in an earlier PR mostly because only the inline mode added any ui APIs at all but perhaps it makes more sense to have it as required. Isn't that a breaking change as well though? Not sure how strict we are with that.